hybig/browse.py

```diff
@@ -31,16 +31,136 @@
     remove_alpha,
 )
-from harmony_browse_image_generator.exceptions import HyBIGError
-from harmony_browse_image_generator.sizes import (
+from hybig.exceptions import HyBIGError
+from hybig.sizes import (
     GridParams,
     create_tiled_output_parameters,
     get_target_grid_parameters,
 )
 
 
+def create_browse(
+    source_tiff: str,
+    params: dict = None,
+    palette: str | ColorPalette | None = None,
+    logger: Logger = None,
+) -> list[tuple[Path, Path, Path]]:
+    """Create browse imagery from an input geotiff.
+
+    This is the exposed library function to allow users to create browse images
+    from the hybig library. It parses the input params and constructs the
+    correct Harmony input structure [Message.Format] to call the service's
+    entry point create_browse_imagery.
+
+    Output images are created and deposited into the input GeoTIFF's directory.
+
+    Args:
+        source_tiff: str, location of the input geotiff to process.
+
+        params: [dict | None], A dictionary with the following keys:
+
+            mime: [str], MIME type of the output image (default: 'image/png').
+                  any string that contains 'jpeg' will return a jpeg image,
+                  otherwise create a png.
+
+            crs: [dict | None], Target image's Coordinate Reference System.
+                 A dictionary with 'epsg', 'proj4' or 'wkt' key.
+
+            scale_extent: [dict | None], Scale Extents for the image. This dictionary
+                contains "x" and "y" keys each whose value which is a dictionary
+                of "min", "max" values in the same units as the crs.
+                e.g.: { "x": { "min": 0.5, "max": 125 },
+                        "y": { "min": 52, "max": 75.22 } }
+
+            scale_size: [dict | None], Scale sizes for the image.  The dictionary
+                contains "x" and "y" keys with the horizontal and veritcal
+                resolution in the same units as the crs.
+                e.g.: { "x": 10, "y": 10 }
+
+            height: [int | None], height of the output image in gridcells.
+
+            width: [int | none], width of the output image in gridcells.
+
+        palette: [str | ColorPalette | none], either a URL to a remote color palette
+             that is fetched and loaded or a ColorPalette object used to color
+             the output browse image. If not provided, a grayscale image is
+             generated.
+
+        logger: [Logger | None], a configured Logger object. If None a default
+             logger will be used.
+
+    Note:
+      if supplied, scale_size, scale_extent, height and width must be
+      internally consistent.  To define a valid output grid:
+            * Specify scale_extent and 1 of:
+              * height and width
+              * scale_sizes (in the x and y horizontal spatial dimensions)
+            * Specify all three of the above, but ensure values are consistent
+              with one another.
```

Comment:
I'm (now fully) tempted to remove this option and tell them to specify two, because it's hard to describe what makes a grid internally consistent without just pointing to the tests that would get run if you were using the harmony-service-lib.

Reply:
Or I could just do this:
bdde01e